torch/testing/_internal/distributed/distributed_utils.py

```diff
@@ -0,0 +1,62 @@
+from contextlib import contextmanager
+from datetime import timedelta
+from functools import (
+    partial,
+    wraps,
+)
+
+import torch.distributed as dist
+
+class MockProcessGroup(dist.ProcessGroup):
```

Comment:
There's a DummyProcessGroup that looks very similar to this `MockProcessGroup`, maybe we can reuse that? https://github.com/pytorch/pytorch/blob/87ab665ba6f42565c48619e1df75e23eee707e98/test/distributed/test_c10d_common.py#L1178

Reply:
I considered it, but test_c10d_common DummyProssGroup has a bunch of test specific implementations that doesn't belong to common testing infra.